blueocean-indexing/src/main/java/io/jenkins/blueocean/indexing/RunLoaderImpl.java

```diff
@@ -0,0 +1,38 @@
+package io.jenkins.blueocean.indexing;
+
+import com.google.inject.Inject;
+import hudson.Extension;
+import hudson.model.Job;
+import io.jenkins.blueocean.commons.ServiceException;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.rest.model.BlueRun;
+import io.jenkins.blueocean.service.embedded.EmbeddedRunLoader;
+import org.apache.lucene.search.MatchAllDocsQuery;
+import org.apache.lucene.search.Sort;
+import org.apache.lucene.search.SortField.Type;
+import org.apache.lucene.search.SortedNumericSortField;
+
+import java.io.IOException;
+
+import static io.jenkins.blueocean.indexing.Fields.sortField;
+
+@Extension
+public class RunLoaderImpl extends EmbeddedRunLoader {
+
+    @Inject
+    IndexService index;
+
+    @Override
+    public Iterable<BlueRun> getRuns(Job job, Link parent) {
+        Index<BlueRun> runIndex = index.getIndex(job);
+        try {
+            return runIndex.query(
+                new MatchAllDocsQuery(),
+                100, new Sort(new SortedNumericSortField(sortField(BlueRun.START_TIME), Type.LONG, true)),
```

Comment:
@vivek need to figure out how we can make this iterable page through hits returned via the query so we can remove the limit here. I am sure there is something todo this in Lucene but haven't had time to investigate.